### PR TITLE
Fix empty and zero width/height values

### PR DESCRIPTION
**Pull request by martijngastkemper**
- Fix Sharp error when zero is given
- Remove hexadecimal width / height value support
- Add tests for empty value

### CODE REVIEW

**Review by harmenjanssen — APPROVED** [review]
LGTM!

File: util/resizeImage.js [hunk 1]
```diff
@@ -30,10 +30,17 @@ function resizeImage(
   const resizeOptions = { fit: fit };
 
   if (width) {
-    resizeOptions.width = parseInt(width);
+    width = parseInt(width, 10);
```
Comment (MartijnNieuwenhuizen) [inline util/resizeImage.js]: I'm not sure if the impact is as big as I imagine, but because you're overwriting the value of `width` here, it could have an impact on other locations in this service.
I often solve this issue in a scenario like this with something like:

```js
// Not sure about the fallback value here, should it be height or 0?
const parsedHeight = height ? parseInt(width, 10) : height;
const parsedHeight = height ? parseInt(width, 10) : 0;

if (parsedHeight > 0) {
      resizeOptions.width = width;
}
```
This also reduces the nested if statements which is nice for future development.

**Review by MartijnNieuwenhuizen — APPROVED** [review]
LGTM!

File: util/resizeImage.js (hunk 1 above)
Comment (martijngastkemper) [reply util/resizeImage.js]: Overwriting a variable inside a function doesn't break out of its scope:

```javascript
const width = "500";
function parseWidth(width) {
  width = parseInt(width);
  return width;
}
console.log(parseWidth(width)); // 500
console.log(width); // "500"
```

But putting the parsed data in a new variable is a good idea.
